src/program.hpp

```diff
@@ -556,9 +557,19 @@ struct cvk_program : public _cl_program, api_object<object_magic::program> {
         return m_binary.constant_data_buffer();
     }
 
-    bool has_region_offset() const {
-        return push_constant(pushconstant::region_offset) &&
-               push_constant(pushconstant::region_group_offset);
+    bool options_allow_split_region(std::string options) {
+        return options.find("-cl-std=CL2.0") != std::string::npos ||
+               options.find("-cl-std=CL3.0") != std::string::npos ||
+               options.find("cl-arm-non-uniform-work-group-size") !=
+                   std::string::npos;
```

Comment:
This doesn't exactly mirror the set of conditions in clspv. It doesn't feel great to hardcode the same set of conditions here. Maybe there is a case to have clspv pass the information in the SPIR-V (i.e. this program was compiled with support for non-uniform NDRanges)? Maybe we should always enable support for non-uniform NDRanges instead of splitting opportunistically? We're definitely going to need more tests to cover all the cases as well :).

Reply:
I feel like always enabling the non-uniform NRRanges is the right move here.
It will be like that anyway when the default version will be 3.0.

Reply:
Yes, we'll have that on by default once we switch to 3.0 (we'll need to rework the clspv flags so the only way to do this is not a deprecated Arm extension). There are ways to optimise further by having multiple variants of the compiled programs but we can treat that as an optimisation for later. Let's always enable non-uniform NDRanges for now. I'll capture options for further optimisations as part of merging this PR.